### ISSUE TITLE
Code Style settings not exported/imported

### ISSUE THREAD

**Issue by miloush** [dotnet/roslyn#14810]
**Version Used**: 2.0.0.61025

**Steps to Reproduce**:
1. Change Code Style preferences and severity.
2. Export settings.
3. Change them again.
4. Import settings.

**Expected Behavior**: Settings restored to exported values.

**Actual Behavior**: Settings left intact by import.

**Comment by jasonmalinowski** [comment]
@jmarolf @dpoeschl We've made a lot of changes around here, any expectation that this works now?

**Comment by jmarolf** [comment]
@jasonmalinowski these are the only settings that are persisted from code styles:

```XML
<UserSettings>
  <ApplicationIdentity version="15.0"/>
  <ToolsOptions>
    <ToolsOptionsCategory name="TextEditor" RegisteredName="TextEditor">
      <ToolsOptionsSubCategory name="CSharp-Specific" RegisteredName="CSharp-Specific" PackageName="CSharpPackage">
        <PropertyValue name="Style_UseImplicitTypeWhereApparent">&lt;CodeStyleOption SerializationVersion="1" Type="Boolean" Value="true" DiagnosticSeverity="Info" /&gt;</PropertyValue>
        <PropertyValue name="Style_QualifyMethodAccess">&lt;CodeStyleOption SerializationVersion="1" Type="Boolean" Value="true" DiagnosticSeverity="Info" /&gt;</PropertyValue>
        <PropertyValue name="Style_PreferIntrinsicPredefinedTypeKeywordInDeclaration">&lt;CodeStyleOption SerializationVersion="1" Type="Boolean" Value="false" DiagnosticSeverity="Info" /&gt;</PropertyValue>
        <PropertyValue name="Style_UseImplicitTypeForIntrinsicTypes">&lt;CodeStyleOption SerializationVersion="1" Type="Boolean" Value="true" DiagnosticSeverity="Info" /&gt;</PropertyValue>
        <PropertyValue name="Style_PreferIntrinsicPredefinedTypeKeywordInMemberAccess">&lt;CodeStyleOption SerializationVersion="1" Type="Boolean" Value="false" DiagnosticSeverity="Info" /&gt;</PropertyValue>
        <PropertyValue name="Style_QualifyEventAccess">&lt;CodeStyleOption SerializationVersion="1" Type="Boolean" Value="true" DiagnosticSeverity="Info" /&gt;</PropertyValue>
        <PropertyValue name="Style_QualifyPropertyAccess">&lt;CodeStyleOption SerializationVersion="1" Type="Boolean" Value="true" DiagnosticSeverity="Info" /&gt;</PropertyValue>
        <PropertyValue name="Style_UseImplicitTypeWherePossible">&lt;CodeStyleOption SerializationVersion="1" Type="Boolean" Value="true" DiagnosticSeverity="Info" /&gt;</PropertyValue>
        <PropertyValue name="Style_QualifyFieldAccess">&lt;CodeStyleOption SerializationVersion="1" Type="Boolean" Value="true" DiagnosticSeverity="Info" /&gt;</PropertyValue>
      </ToolsOptionsSubCategory>
    </ToolsOptionsCategory>
  </ToolsOptions>
</UserSettings>
```

Looking at [AutomationObject](https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs) it appears we've missed several options.  All of these need to be on automation object to export/import correctly

CodeStyleOptions In Automation Object:
 - QualifyFieldAccess 
 - QualifyPropertyAccess 
 - QualifyMethodAccess 
 - QualifyEventAccess 
 - PreferIntrinsicPredefinedTypeKeywordInDeclaration 
 - PreferIntrinsicPredefinedTypeKeywordInMemberAccess
 
CodeStyleOptions **Not** in Automation Object:
 - PreferThrowExpression
 - PreferObjectInitializer
 - PreferCollectionInitializer
 - PreferCoalesceExpression
 - PreferNullPropagation
 - PreferInlinedVariableDeclaration
 - PreferExplicitTupleNames
 - PreferExplicitTupleNames

CSharpCodeStyleOptions In Automation Object:
 - UseVarWhenDeclaringLocals 
 - UseImplicitTypeForIntrinsicTypes 
 - UseImplicitTypeWhereApparent 
 - UseImplicitTypeWherePossible 

CSharpCodeStyleOptions **Not** In Automation Object:
 - PreferConditionalDelegateCall
 - PreferPatternMatchingOverAsWithNullCheck
 - PreferPatternMatchingOverIsWithCastCheck
 - PreferExpressionBodiedConstructors
 - PreferExpressionBodiedMethods
 - PreferExpressionBodiedOperators
 - PreferExpressionBodiedProperties
 - PreferExpressionBodiedIndexers
 - PreferExpressionBodiedAccessors
 - PreferBraces

**Comment by Pilchie** [comment]
Note that we should also update the profiles that ship so that reset all settings actually resets these options.

**Comment by jmarolf** [comment]
Fixed with #16512
